src/InfluxDB.Collector/Diagnostics/CollectorLog.cs

```diff
@@ -1,15 +1,54 @@
 ﻿using System;
-using System.IO;
+using System.Collections.Generic;
+using InfluxDB.Collector.Util;
 
 namespace InfluxDB.Collector.Diagnostics
 {
     public static class CollectorLog
     {
-        public static TextWriter Out { get; set; } = Console.Error;
+        private static readonly HashSet<Action<string, Exception>> ErrorHandlers = new HashSet<Action<string, Exception>>();
+        private static readonly object SyncRoot = new object();
 
-        public static void WriteLine(string format, params object[] args)
+        /// <summary>
+        /// Registers an error handler. Errors reported may not neccessarily always correspond with an exception. That is to say, when the callback is invoked, the exception may be null.
+        /// To unregister a handler, simply dispose of the IDisposable returned from this method
+        /// The order in which handlers are invoked is not defined
+        /// </summary>
+        /// <param name="errorHandler">A func accepting a string and an exception. Note that exception may be null when invoked</param>
+        /// <returns>An IDispoable which when explicitly disposed, will unnregister the handler</returns>
+        public static IDisposable RegisterErrorHandler(Action<string, Exception> errorHandler)
         {
```

Comment:
Good call, don't typically work on public libraries so tend be a bit slack on that kind of stuff. Will fix.